src/blocks/gallery-carousel/edit.js

```diff
@@ -131,277 +117,261 @@ class GalleryCarouselEdit extends Component {
-				>
-					<div className={ className }>
-						<div className={ innerClasses }>
-							<Flickity
-								className={ flickityClasses }
-								disableImagesLoaded={ true }
-								flickityRef={ ( c ) => this.flkty = c }
-								options={ flickityOptions }
-								reloadOnUpdate={ true }
-								updateOnEachImageLoad={ true }
-							>
-								{ images.map( ( img, index ) => {
-									const ariaLabel = sprintf(
-										/* translators: %1$d is the order number of the image, %2$d is the total number of images */
-										__( 'image %1$d of %2$d in gallery', 'coblocks' ),
-										( index + 1 ),
-										images.length
-									);
-
-									return (
-										<div className="coblocks-gallery--item" role="button" tabIndex="0" key={ img.id || img.url } onKeyDown={ this.onItemClick } onClick={ this.onItemClick }>
-											<GalleryImage
-												url={ img.url }
-												alt={ img.alt }
-												id={ img.id }
-												gutter={ gutter }
-												gutterMobile={ gutterMobile }
-												marginRight={ true }
-												marginLeft={ true }
-												isSelected={ isSelected && this.state.selectedImage === index }
-												onRemove={ this.onRemoveImage( index ) }
-												onSelect={ this.onSelectImage( index ) }
-												setAttributes={ ( attrs ) => this.setImageAttributes( index, attrs ) }
-												caption={ img.caption }
-												aria-label={ ariaLabel }
-												supportsCaption={ false }
-												supportsMoving={ false }
-												imageIndex={ index }
-												replaceImage={ this.replaceImage }
-											/>
-										</div>
-									);
-								} ) }
-							</Flickity>
-						</div>
-					</div>
-				</ResizableBox>
-				{ thumbnails &&
-					<div className={ className }>
-						<div
-							className={ innerClasses }
-							style={ navStyles }
+	);
+
+	const flickityClasses = classnames(
+		'has-carousel',
+		`has-carousel-${ gridSize }`, {
+			'has-aligned-cells': alignCells,
+			[ `has-margin-bottom-${ gutter }` ]: thumbnails && gutter > 0,
+			[ `has-margin-bottom-mobile-${ gutterMobile }` ]: thumbnails && gutterMobile > 0,
+			[ navForClass ]: thumbnails,
+		}
+	);
+
+	const navClasses = classnames(
+		'carousel-nav', {
+			[ `has-margin-top-${ gutter }` ]: gutter > 0,
+			[ `has-margin-top-mobile-${ gutterMobile }` ]: gutterMobile > 0,
+			[ `has-negative-margin-left-${ gutter }` ]: gutter > 0,
+			[ `has-negative-margin-left-mobile-${ gutterMobile }` ]: gutterMobile > 0,
+			[ `has-negative-margin-right-${ gutter }` ]: gutter > 0,
+			[ `has-negative-margin-right-mobile-${ gutterMobile }` ]: gutterMobile > 0,
+		}
+	);
+
+	const flickityOptions = {
+		draggable: false,
+		pageDots: true,
+		prevNextButtons: true,
+		wrapAround: true,
+		autoPlay: false,
+		cellAlign: alignCells ? 'left' : 'center',
+		arrowShape: {
+			x0: 10,
+			x1: 60, y1: 50,
+			x2: 65, y2: 45,
+			x3: 20,
+		},
+		responsiveHeight,
+		thumbnails,
+	};
+
+	const navOptions = {
```

Comment:
would there be any benefit to memoizing some of these objects using `useMemo`?

Reply:
Because there are not any expensive computations occurring when setting the navOptions, I do not think it will benefit us here to memorize this value.

Reply:
If we could improve the performance of the Block by using memo more frequently across all/most of the instantiated objects it might be worth looking into. That said, I think that would be a good exploration for another PR.